src/main/java/de/tum/in/www1/artemis/domain/settings/ide/UserIdeMappingId.java

```diff
@@ -0,0 +1,48 @@
+package de.tum.in.www1.artemis.domain.settings.ide;
+
+import java.io.Serializable;
+
+import jakarta.persistence.IdClass;
+
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+
+@IdClass(UserIdeMappingId.class)
+public class UserIdeMappingId implements Serializable {
```

Comment:
Can you make this an inner class of `UserIdeMapping`?

Reply:
do you only mean a local inner class or also use rather `@ Embeddable` like Code Rabbit suggested. 
Is there any Code guideline concerning this?

Reply:
I don't think we have any guide lines here, but I thought about making it similar to `CompetencyUserId` which is actually also @ Embeddable